src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -81,107 +84,243 @@ public static Path yamlQuarkiverseMetadataPath(String version) {
-                Collectors.toUnmodifiableMap(Map.Entry::getKey, e -> config.localized().get(e.getKey().code).webUri()));
+        HashMap<Language, URI> languageUriMap = new HashMap<>(localizedSites.entrySet().stream().collect(
+                Collectors.toMap(Map.Entry::getKey, e -> config.localized().get(e.getKey().code).webUri())));
+        languageUriMap.put(Language.ENGLISH, config.webUri());
+        this.localizedSiteUris = Collections.unmodifiableMap(languageUriMap);
         this.failureCollector = failureCollector;
+
+        Map<Language, GitCloneDirectory> all = new HashMap<>(localizedSites);
+        all.put(Language.ENGLISH, mainRepository);
+        this.allSites = Collections.unmodifiableMap(all);
+
+        validateRepositories(mainRepository, localizedSites, failureCollector);
+    }
+
+    private static void validateRepositories(
+            GitCloneDirectory mainRepository,
+            Map<Language, GitCloneDirectory> localizedSites,
+            FailureCollector failureCollector) {
+        ObjectId latestMainRev = mainRepository.currentSourcesLatestHash();
+        for (Map.Entry<Language, GitCloneDirectory> localized : localizedSites.entrySet()) {
+            localized.getValue().currentUpstreamSubmoduleSourcesHash().ifPresentOrElse(commitHash -> {
+                if (latestMainRev.equals(commitHash)) {
+                    // means that the localized repo points to the latest main one and we can ignore further checks
+                    return;
+                }
+
+                // Otherwise let's see when the localized site was last synced,
+                // by looking at when that latest commit was committed:
+                Instant lastSync = GitUtils.committedInstant(mainRepository.git(), commitHash);
+                // Sync happens each week, so we can have
+                //  1. a commit to main repo on Monday
+                //  2. no further commits during the week
+                //  3. sync on the weekend (Sunday?)
+                //  4. commits to main repo
+                //  at this point the difference between latest commit in localized site and current date can be
+                //  more than 7 days; so we give it 2 weeks period before we start considering that there's a sync problem:
+                Duration duration = Duration.between(lastSync, Instant.now());
+                if (duration.compareTo(Duration.ofDays(14)) > 0) {
+                    failureCollector.warning(
+                            FailureCollector.Stage.TRANSLATION,
+                            "Localized repository '" + localized.getKey() + "' is out of sync for " + duration);
+                }
+            }, () -> {
+                Log.debug(
+                        "Repository for a localized site '" + localized.getKey()
+                                + "' does not have an \"upstream\" directory. Site will be considered up-to-date. ");
```

Comment:
This probably warrants a warning... ? Unless it's a valid situation, but apart from tests I don't think it is. And I'm fine with warnings in tests.